### PR TITLE
Projects view

### PR DESCRIPTION
**Pull request by sudoFerraz**
## Description

This PR adds the projects view, which merges together single repository teams, and multi repositories team into a single view and detail page. 
This view is not available on the dashboard menu yet

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Screenshots (if appropriate):
image

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/app/(dashboard)/projects/[projectId]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+  const [userTooltipDto, setUserTooltipDto] = useState<UserTooltipDto[]>([]);
+  const [hasContributionRequest, setHasContributionRequest] = useState(false);
+  const [contributionCalculation, setContributionCalculation] =
+    useState<ContributionCalculation>();
+  const [pollingCount, setPollingCount] = useState<number>(0);
+
+  const router = useRouter();
+
+  useEffect(() => {
+    setIsLoading(true);
```
Comment (AlecErasmus) [inline src/app/(dashboard)/projects/[projectId]/page.tsx]: This is not needed right? The default value is already true

File: src/app/(dashboard)/projects/[projectId]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    const { data } = await axios.get("/api/teams/overview?team_id=" + teamId);
+    if (data.success) {
+      setOverview(data.overview);
+    }
+  };
+
+  useEffect(() => {
+    if (userCredDtos.length > 0) {
+      let index = 0;
+      const userTooltipDtoArray = userCredDtos.map((userCredDto) => {
```
Comment (AlecErasmus) [inline src/app/(dashboard)/projects/[projectId]/page.tsx]: You can use map((obj, index). No need to keep your own counter

File: src/app/(dashboard)/projects/page.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import { Separator } from "@/components/ui/separator";
+import axios from "axios";
+import { Plus } from "lucide-react";
+import { useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+import { Team } from "@prisma/client";
+import { TextGenerateEffect } from "@/components/ui/text-generate-effect";
+import { ProjectCard } from "@/components/projects/projectCard";
+
+const breadcrumbItems = [{ title: "Projects", link: "/dashboard/projects" }];
```
Comment (AlecErasmus) [inline src/app/(dashboard)/projects/page.tsx]: We usually put this inside the export default function TeamsPage(). Maybe for consistency keep it the same as the rest?

File: src/app/(dashboard)/projects/page.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        <div className="flex items-start justify-between">
+          <Heading
+            title={`Projects`}
+            description={`Manage your projects or create a new one`}
+          />
+          <Button
+            className="text-xs md:text-sm"
+            // TODO
+            // onClick={() => router.push(`/repositories/new`)}
+          >
```
Comment (AlecErasmus) [inline src/app/(dashboard)/projects/page.tsx]: we keep this right?

File: src/app/(dashboard)/projects/[projectId]/page.tsx (hunk 1 above)
Comment (sudoFerraz) [reply src/app/(dashboard)/projects/[projectId]/page.tsx]: Hmm good point, let me test it out

File: src/app/(dashboard)/projects/page.tsx (hunk 4 above)
Comment (sudoFerraz) [reply src/app/(dashboard)/projects/page.tsx]: Yeah, I'm unsure still if we want to keep this button there tbh, as the "Add repositories" option will be on the left menu bar

**Review by AlecErasmus — APPROVED** [review]

**Pull request merged**